rules/disposableemails/classes/rule.php

```diff
@@ -58,18 +58,43 @@ public function post_data_check(array $data): ?rule_check_result {
 
         $email = $data['email'];
 
+        // Proper validation for email addresses should be handled by the
+        // form instance, but we'll do a basic check and return null if this
+        // isn't actually an email address.
         if (strpos($email, '@') === false) {
-            return new rule_check_result(true, 'Invalid email address', 0);
+            return null;
         }
```

Comment:
We could also use Moodle's sanitisation methods to check for it? So it's consistent in Moodle's eyes…

e.g. `if (clean_param($email, PARAM_EMAIL) !== $email) {`

Reply:
Yeah I'm happy to use this instead, might as well check it properly

Reply:
To simplify I've used Moodle's `validate_email()` instead as it's what `clean_param` uses under the hood anyway